repos/dir.props

```diff
@@ -72,7 +72,7 @@
     <EnvironmentVariables Include="DOTNET_HOST_PATH=$(DotNetCliToolDir)dotnet" />
     <!-- _InitializeDotNetCli is used by websdk and templating to decide whether to init the SDK -->
     <EnvironmentVariables Include="_InitializeDotNetCli=$(DotNetCliToolDir)" />
-    <EnvironmentVariables Include="_InitializeToolset=$(ProjectDir)Tools/source-built/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj" Condition="'$(RepositoryName)' != 'arcade'" />
+    <EnvironmentVariables Include="_InitializeToolset=$(ProjectDir)Tools/source-built/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj" Condition="'$(RepositoryName)' != 'arcade' AND '$(RepositoryName)' != 'sourcelink'" />
```

Comment:
I don't think the `dir.props` should condition on repo names... seems like it should use a property name focused on the intent that `proj` files can set. When I see this, it seems analogous to having this in a base class:

```C#
if (!(foo is Arcade) && !(foo is SourceLink)) { addEnv(...); }
```

Looks like this is existing (so I suppose non-blocking) but something to consider.